File: src/main/java/strategy/Remit.java
package strategy;

import strategy.fee.Fee;

public abstract class Remit {
    private final Fee fee;

    public Remit(Fee fee) {
        this.fee = fee;
    }

    public abstract String createTransferMessage();

    public double calculateAmount(double amount) {
        return amount + calculateFee(amount);
    }

    private double calculateFee(double amount) {
        return this.fee.calculate(amount);
    }
}

File: kotlin/src/main/kotlin/strategy/Remit.kt
package strategy

import strategy.fee.Fee

abstract class Remit(private val fee: Fee) {
    protected abstract fun createTransferMessage(): String

    fun calculateAmount(amount: Double): Double {
        return amount + calculateFee(amount)
    }

    private fun calculateFee(amount: Double): Double {
        return fee.calculate(amount)
    }
}